rust/lance-linalg/build.rs

```diff
@@ -0,0 +1,34 @@
+fn main() {
+    println!("cargo:rerun-if-changed=src/simd/f16.c");
+
+    if cfg!(all(target_os = "macos", target_feature = "neon")) {
+        cc::Build::new()
+            .compiler("clang")
+            .file("src/simd/f16.c")
+            .flag("-mcpu=apple-m1")
+            .flag("-ffast-math")
+            .flag("-O3")
+            .flag("-Wall")
+            .flag("-Werror")
+            .flag("-Wextra")
+            .flag("-Wpedantic")
+            .compile("f16");
+    }
+
+    if cfg!(all(target_os = "linux", feature = "avx512fp16")) {
+        // No fp16 without AVX512fp16
+        cc::Build::new()
+            .compiler("clang")
+            .std("c17")
+            .file("src/simd/f16.c")
+            .flag("-march=sapphirerapids")
+            .flag("-ffast-math")
```

Comment:
This makes the math non-deterministic. Probably okay, but is the perf gain worth it?

Reply:
Oh i was hoping it gives a reduction at the end.